src/endpoint-client.ts

```diff
@@ -122,6 +128,12 @@ export class EndpointClient {
 		if (this.logger.isDebugEnabled()) {
 			this.logger.debug(`making axios request: ${JSON.stringify(axiosConfig)}`)
 		}
+		if (options?.dryRun) {
+			if (options.dryRunReturnValue) {
+				return options.dryRunReturnValue
+			}
+			throw new Error('skipping request; dry run mode')
```

Comment:
Can we just return nothing and log instead of throwing an Error? Maybe there something better about throwing an error I am not seeing.

Reply:
We need to return a `T` for the compiler to be happy. Since we don't know what `T` is at runtime, we can't fake one like one could in Java.

Reply:
(That's why I included the `dryRunReturnValue` since the user _will_ know what T is and can supply a good default if they want to.)

Reply:
Ahh, makes sense!